components/reporters/slack/internal/reporter/slack/client.go

```diff
@@ -0,0 +1,140 @@
+package slack
+
+import (
+	"context"
+	"log/slog"
+	"net/http"
+
+	"github.com/go-errors/errors"
+	"github.com/slack-go/slack"
+	componentlogger "github.com/smithy-security/smithy/sdk/logger"
+)
+
+type (
+	client struct {
+		client    *slack.Client
+		channelID string
+	}
+
+	// Config allows to customise the client config.
+	Config struct {
+		SlackToken     string
```

Comment:
this Config struct is used as `SlackClientConfig` in the transformer which means that the fields are going to be referenced as `r.SlackClientConfig.SlackToken` which is smurfy. we could simplify this to just token and I don't think anyone would be confused